ledger-api/grpc-definitions/com/daml/ledger/api/v1/transaction_service.proto

```diff
@@ -178,3 +183,33 @@ message GetLedgerEndResponse {
   LedgerOffset offset = 1;
 }
 
+message GetExerciseEventsRequest {
+  // Must correspond to the ledger ID reported by the Ledger Identification Service.
+  // Must be a valid LedgerString (as described in ``value.proto``).
+  // Optional
+  string ledger_id = 1;
+
+  // Beginning of the requested ledger section.
+  // This offset is exclusive: the response will only contain transactions whose offset is strictly greater than this.
+  // Required
+  LedgerOffset begin = 2;
+
+  // End of the requested ledger section.
+  // This offset is inclusive: the response will only contain transactions whose offset is less than or equal to this.
+  // Optional, if not set, the stream will not terminate.
+  LedgerOffset end = 3;
+
+  // Requesting parties with template filters.
+  // Template filters must be empty for GetTransactionTrees requests.
+  // Required
+  TransactionExerciseFilter filter = 4;
+
+  // If enabled, values served over the API will contain more information than strictly necessary to interpret the data.
+  // In particular, setting the verbose flag to true triggers the ledger to include labels for record fields.
+  // Optional
+  bool verbose = 5;
```

Comment:
The JSON API uses this at the moment. I think Daml Script might also still rely on it in some places but I could be wrong there. We have pretty generic functionality to add that to a value that doesn’t have it so I don’t see a huge issue.

Reply:
Clients that might want this functionality:
- Generic clients like Navigator, i.e., those that do not have a compile time dependency on a package.
- Clients that do not use code generation because they are written in some non-standard language.

Implementing the backfilling in the participant is easy, implementing it on the client side is too difficult for an average developer.

Reply:
OK thx. Let's keep it in then.